### ISSUE TITLE
[interface-name-prefix] <how to extend window(dom) property>

### ISSUE THREAD

**Issue by Mr-jiangzhiguo** [typescript-eslint/typescript-eslint#969]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/interface-name-prefix": ["error",{ "prefixWithI": "always" }],
  }
}
```

```TS
// your repro code case
declare global {
  interface Window {
    postTo: string
  }
  interface IAnyObj {
    [propName: string]: any
  }
}

export {}
```

**Expected Result**
use `interface-name-prefix` rule,
while extend `window`(`dom`),I hope no error info
**Actual Result**
![image](https://user-images.githubusercontent.com/22312092/64753810-2ae48080-d557-11e9-94dd-e78a40d613f3.png)

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `^2.1.0` |
| `@typescript-eslint/parser`        | `^2.1.0` |
| `TypeScript`                       | `3.6.2` |
| `ESLint`                           | `^6.3.0` |
| `node`                             | `10.15.3` |
| `npm`                              | `6.4.1` |

**Comment by a-tarasyuk** [comment]
I think it is correct behavior for this rule because the rule doesn't check if this global or local definition. TS provides many interfaces without `I` prefix. To fix this you can disable rule for this interface or for the entire file.

```TS
// your repro code case
declare global {
  // eslint-disable-next-line @typescript-eslint/interface-name-prefix
  interface Window {
    postTo: string
  }
  interface IAnyObj {
    [propName: string]: any
  }
}
export {}
```

**Comment by Mr-jiangzhiguo** [comment]
> I think it is correct behavior for this rule because the rule doesn't check if this global or local definition. TS provides many interfaces without `I` prefix.
> 
> To fix this you can disable rule for this interface or for the entire file.
> 
> ```ts
> // your repro code case
> declare global {
>   // eslint-disable-next-line @typescript-eslint/interface-name-prefix
>   interface Window {
>     postTo: string
>   }
>   interface IAnyObj {
>     [propName: string]: any
>   }
> }
> export {}
> ```

thx
